### ISSUE TITLE
 what's the training result on imagenet when training from scratch ?

### ISSUE THREAD

**Issue by anxu829** [szq0214/MEAL-V2#12]
Hi @MingSun-Tse,
i have noticed that you said you may train you distillation from scratch (random initial) on imagenet ,
i am wondering whats your training result because i want to use your method to train on my own dataset , while all i have is a large model train on this dataset .  should i train this model on resnet50 firstly and than use your code to finetune or i can directly use your code to distillation exists model ?

**Comment by szq0214** [comment]
@anxu829 you can pre-train your small model first and then fine-tune with our code. This will be very helpful.

I tried the experiment from scratch, if not using the good initialization, hard label seems important to obtain similar performance (~1% lower than the reported accuracy). I think they are equivalent since the good initialization is trained from hard labels. I will give more details in the long version of our paper and please stay tuned.

**Comment by anxu829** [comment]
> @anxu829 you can pre-train your small model first and then fine-tune with our code. This will be very helpful.
> 
> I tried the experiment from scratch, if not using the good initialization, hard label seems important to obtain similar performance (~1% lower than the reported accuracy). I think they are equivalent since the good initialization is trained from hard labels. I will give more details in the long version of our paper and please stay tuned.

thank you for your reply , i will train on hard label firstly

**Comment by woshichase** [comment]
> 
> 
> @anxu829 you can pre-train your small model first and then fine-tune with our code. This will be very helpful.
> 
> I tried the experiment from scratch, if not using the good initialization, hard label seems important to obtain similar performance (~1% lower than the reported accuracy). I think they are equivalent since the good initialization is trained from hard labels. I will give more details in the long version of our paper and please stay tuned.

Hi, may I know how many epochs does it take for training from scratch with hard label, achieving ~1% performance lower than reported?

**Comment by szq0214** [comment]
@woshichase hard label is only used to initialize the parameters, we train with it following the standard protocol (90 epochs), then finetuning with our framework (180 epochs).

**Comment by woshichase** [comment]
> 
> 
> @woshichase hard label is only used to initialize the parameters, we train with it following the standard protocol (90 epochs), then finetuning with our framework (180 epochs).

Thanks~ But I am still wondering, what are the results of your distillation approach if trained from scratch?  You said you have tried the experiment from scratch, but it seems it's still a finetuning process from your description.

**Comment by woshichase** [comment]
> 
> 
> @woshichase hard label is only used to initialize the parameters, we train with it following the standard protocol (90 epochs), then finetuning with our framework (180 epochs).

@szq0214 Thanks, It still seems a finetuning process. You said you have tried the expriment from scratch, so I am still wondering what are the results of your distillaton approach from scratch?

**Comment by szq0214** [comment]
@woshichase yeah, it can be regarded as **hard label pre-training + soft label fine-tuning**. We obtained 79.48% still without any tricks, this is a very preliminary result as I just took a quick run. It seems from the validation trend that more epochs can continue boosting this accuracy, so I think some small adjustments could make this result better.

**Comment by woshichase** [comment]
> @woshichase yeah, it can be regarded as **hard label pre-training + soft label fine-tuning**. We obtained 79.48% still without any tricks, this is a very preliminary result as I just took a quick run. It seems from the validation trend that more epochs can continue boosting this accuracy, so I think some small adjustments could make this result better.

@szq0214  Thank you~ Seen from the current results, I suppose the proposed distillation approach is suitable for post-training to further enchance the performance. But it exists no much superiority to use it for direct training?

**Comment by szq0214** [comment]
@woshichase I don't think so. You can see that the current settings we used for distillation are somewhat unique like we did not use weight decay. This may be helpful if the parameters are initialized in good status, but may be incorrect if the weights are randomly initialized. So the thing we can confirm currently is that the soft labels are fairly powerful to obtain competitive results (80.67%), but how to use them is not yet determined. We provided a demonstration in MEAL V2 that **hard label pre-training + soft label fine-tuning** is effective to utilize such powerful soft labels, while if hard label pre-training is not existing, the configuration might need to be slightly adjusted to fit the circumstance, for example, using weight decay, warmup, cosine decay learning rate, etc. to unleash the potential of the soft labels.

**Comment by woshichase** [comment]
> 
> 
> @woshichase I don't think so. You can see that the current settings we used for distillation are somewhat unique like we did not use weight decay. This may be helpful if the parameters are initialized in good status, but may be incorrect if the weights are randomly initialized. So the thing we can confirm currently is that the soft labels are fairly powerful to obtain competitive results (80.67%), but how to use them is not yet determined. We provided a demonstration in MEAL V2 that **hard label pre-training + soft label fine-tuning** is effective to utilize such powerful soft labels, while if hard label pre-training is not existing, the configuration might need to be slightly adjusted to fit the circumstance, for example, using weight decay, warmup, cosine decay learning rate, etc. to unleash the potential of the soft labels.

@szq0214 I understand its effectiveness for such a two-stage training paradigm. However it's sometimes not handy to change the training phase in practice.  So I much look forward to seeing your results of direct training after making adjustments of the hyper-parameters. Thank you.
